### PR TITLE
ci(cypress): Added shipping address test cases for PML

### PR DESCRIPTION
**Pull request by likhinbopanna**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
Added shipping address test cases for payment method list in cypress automation

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context
Shipping address test cases wasn't present in the cypress automation 

## How did you test it?
Cypress
- PML alone
image

- Stripe
image

- Cybersource
image

## Checklist

- [ ] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: cypress-tests/cypress/e2e/PaymentMethodListTest/00000-PaymentMethodListTests.cy.js [hunk 1]
```diff
@@ -423,4 +423,163 @@ describe("Payment Method list using Constraint Graph flow tests", () => {
       });
     }
   );
+
+  // Testing for scenario:
+  // MCA1 -> Stripe configured with card credit
+  // MCA1 -> Cybersource configured with card credit = { currency = "USD" }
+  // Payment is done with currency as USD and no billing address
+  // The resultant Payment Method list should have
+  // Stripe and cybersource both for credit
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/PaymentMethodListTest/00000-PaymentMethodListTests.cy.js:432]: You can remove this comment since you have added the same within `context()`.

File: cypress-tests/cypress/e2e/PaymentMethodListTest/00000-PaymentMethodListTests.cy.js [hunk 2]
```diff
@@ -316,7 +316,7 @@ describe("Payment Method list using Constraint Graph flow tests", () => {
         let res_data = data["Response"];
 
         cy.createPaymentIntentTest(
-          create_payment_body_with_currency_country("EUR", "US"),
+          create_payment_body_with_currency_country("EUR", "US", "US"),
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/PaymentMethodListTest/00000-PaymentMethodListTests.cy.js:319]: I have never seen a use case where we pass billing country and shipping country separate and they always remained the same.  
I would suggest you to revert this change.

File: cypress-tests/cypress/e2e/PaymentMethodListUtils/Commons.js [hunk 3]
```diff
@@ -114,7 +114,8 @@ export const bank_redirect_ideal_and_credit_enabled = [
 
 export const create_payment_body_with_currency_country = (
   currency,
-  billingCountry
+  billingCountry,
+  shippingCountry
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/PaymentMethodListUtils/Commons.js:118]: remove `shippingCountry` and rename `billingCountry` to something like, let's say `country`(?) and pass it as a parameter to both `billing` and `shipping` below.

File: cypress-tests/cypress/e2e/PaymentMethodListUtils/Commons.js [hunk 4]
```diff
@@ -146,6 +147,24 @@ export const create_payment_body_with_currency_country = (
...
+      line3: "Ramnagar",
+      city: "Bengaluru",
+      state: "Karnataka",
+      zip: "560011",
+      country: shippingCountry, // Shipping country from parameter
+      first_name: "John",
+      last_name: "Joseph",
+    },
+    phone: {
+      number: "9888888888",
```
Comment (pixincreate) [inline cypress-tests/cypress/e2e/PaymentMethodListUtils/Commons.js:163]:
```suggestion
      number: "9123456789",
```

Change it else where as well (L145 above).

**Review by pixincreate — COMMENTED** [review]
Other than that, LGTM.

File: cypress-tests/cypress/e2e/PaymentMethodListTest/00000-PaymentMethodListTests.cy.js (hunk 2 above)
Comment (likhinbopanna) [reply cypress-tests/cypress/e2e/PaymentMethodListTest/00000-PaymentMethodListTests.cy.js:319]: We have this scenario everywhere , this is one of the use cases for every merchant , so i added this scenario

File: cypress-tests/cypress/e2e/PaymentMethodListUtils/Commons.js (hunk 3 above)
Comment (likhinbopanna) [reply cypress-tests/cypress/e2e/PaymentMethodListUtils/Commons.js:118]: we need shipping and billing separately

File: cypress-tests/cypress/e2e/PaymentMethodListTest/00000-PaymentMethodListTests.cy.js (hunk 2 above)
Comment (pixincreate) [reply cypress-tests/cypress/e2e/PaymentMethodListTest/00000-PaymentMethodListTests.cy.js:299]: Since we've hard coded the value to be `US` for both (billing and shipping), why not go with a single variable?

Comment (likhinbopanna) [reply cypress-tests/cypress/e2e/PaymentMethodListTest/00000-PaymentMethodListTests.cy.js:299]: We have a case where we need to add different values

**Review by pixincreate — DISMISSED** [review]

**Review by pixincreate — APPROVED** [review]

**Review by prajjwalkumar17 — APPROVED** [review]

**Review by Gnanasundari24 — APPROVED** [review]
